README.md

```diff
@@ -1,2 +1,308 @@
+### getVBUSVoltage() ###
+
+Returns the VBUS voltage based on the ADC conversion, this is the input voltage
+
+#### Parameters ####
+
+None.
+
+#### Return Value ####
+
+Integer — The battery voltage in milli volts.
+
+#### Example ####
+
+```squirrel
+local voltage = batteryCharger.getVBUSVoltage();
+server.log("Voltage: " + voltage + "mV");
+```
+### getSystemVoltage() ###
+
+Returns the system voltage based on the ADC conversion, this the output voltage which can be used to power other chips in your application
+
+#### Parameters ####
+
+None.
+
+#### Return Value ####
+
+Integer — The battery voltage in milli volts.
+
+#### Example ####
+
+```squirrel
+local voltage = batteryCharger.getSystemVoltage();
+server.log("Voltage: " + voltage + "mV");
+```
+### getChargingCurrent() ###
+
+Returns the measured current going to the battery 
+#### Parameters ####
+
+None.
+
+#### Return Value ####
+
+Integer — The charging current in milli Amps.
+
+#### Example ####
+
+```squirrel
+local current = batteryCharger.getChargingCurrent();
+server.log("Current: " + current + "mA");
+```
+### getChargingStatus() ###
+
+ Returns the charging status
+
+#### Parameters ####
+
+None.
+
+#### Return Value ####
+
+Integer — Not Charging = 0,  Pre-charge = 1, Fast Charging = 2, Charge Termination Good = 3
+```squirrel
+enum BQ25895M_CHARGING_STATUS{
+    NOT_CHARGING = 0x00, // 0
+    PRE_CHARGE = 0x08, // 1
+    FAST_CHARGING = 0x10, // 2
+    CHARGE_TERMINATION_DONE = 0x18 // 3
+}
+```
+#### Example ####
+
+```squirrel
+local status = charger. getChargingStatus();  
+switch(status) {  
+	case BQ25895M_CHARGING_STATUS.NOT_CHARGING :  
+		// Do something  
+		break;  
+	case BQ25895M_CHARGING_STATUS.PRE_CHARGE :  
+		// Do something  
+		break;  
+	case BQ25895M_CHARGING_STATUS.FAST_CHARGING :  
+		// Do something  
+		break;  
+	case BQ25895M_CHARGING_STATUS.CHARGE_TERMINATION_DONE :  
+		// Do something  
+		break;  
+}
+```
+### getChargerFaults() ###
+
+ Returns the possible charger faults
+
+#### Parameters ####
+
+None.
+
+#### Return Value ####
+Table &mdash; the possible charger faults
+
+| Fault | Type | Description |
+| --- | --- | --- |
+| *watchdogFault* | Bool | `true` if watchdog timer has expired|
+| *boostFault* | Bool | `true` if VBUS overloaded in OTG, VBUS OVP, or battery is too low  |
+| *chrgFault* | Enum | *enumerated type |
+| *battFault* | Bool| `true` if VBAT > VBATOVP |
+| *ntcFault* | Enum | *enumerated type |
+
+*CHRG_FAULT has an enumerated type to match its output.
+```squirrel
+enum BQ25895M_CHARGING_FAULT{
+    NORMAL, // 0
+    INPUT_FAULT, // 1
+    THERMAL_SHUTDOWN, // 2
+    CHARGE_SAFETY_TIMER_EXPIRATION // 3
+}
+```
+*NTC_FAULT has an enumerated type to match its output.
+```squirrel
+enum BQ25895M_NTC_FAULT{
+    NORMAL, // 0
+    TS_COLD, // 1
+    TS_HOT // 2
+}
+```
+
+#### Example ####
+
+```squirrel
+local chargingStatus = batteryCharger.getChargerFaults();
+```
+### reset() ###
+
+Restores the devices default settings
```

Comment:
This method blocks for 1 sec. This needs to be noted here. (Blocking for 1 sec is not ideal - if reset truly needs a full second to execute properly I'd consider making it async and adding a callback.